### PR TITLE
added amplitude peaks script that should be a proc or mini scheduled job :)

### PR DESCRIPTION
**Pull request by mattlianje**
A picture speaks many words 
**testing steps**
`% cd audio-analysis`
`% python aupdate-audio-peaks.py` or `% python aupdate-audio-peaks.py log` <- for local testing, but you'll need a `labelled`table with an ìs_amplitude_peak` boolean column - DDL below
image

```
CREATE TABLE `labelled` (
 `url` varchar(500) NOT NULL,
 `start` int(11) NOT NULL,
 `end` int(11) NOT NULL,
 `commentator` int(11) DEFAULT NULL,
 `crowd` int(11) DEFAULT NULL,
 `isHighlight` int(11) DEFAULT NULL,
 `gameplay` int(11) DEFAULT NULL,
 `updated_at` datetime DEFAULT CURRENT_TIMESTAMP,
 `completed` tinyint(1) DEFAULT '0',
 `priority` int(11) NOT NULL DEFAULT '10',
 `mfcc` text,
 `amplitude` float DEFAULT NULL,
 `pitch` float DEFAULT NULL,
 `word` varchar(500) DEFAULT NULL,
 `subjectivity` float DEFAULT NULL,
 `polarity` float DEFAULT NULL,
 `is_amplitude_peak` tinyint(1) DEFAULT NULL,
 PRIMARY KEY (`url`,`start`,`end`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8
```

### CODE REVIEW

File: audio-analysis/update-audio-peaks.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+clean_labelled_df = labelled_df.rename(columns={'start': 'start_time_s'})
+# Get the list of urls for which amplitude is NOT NULL and is_amplitude_peak is NULL.
+filtered_labelledDF = labelled_df[(labelled_df['amplitude'].notnull()) & (labelled_df['is_amplitude_peak'].isnull())]
+url_list = filtered_labelledDF.url.unique()
+if not url_list:
+    print('All videos already have amplitude peak tags!')
+else:
+    print('Going through videos with is_amplitude_peak tag ...')
+    column_name = 'is_amplitude_peak'
+    for url in tqdm(url_list):
```
Comment (tjlam) [inline audio-analysis/update-audio-peaks.py:38]: what does tqdm do?

File: audio-analysis/update-audio-peaks.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        temp_df = clean_labelled_df[clean_labelled_df['url'] == url].sort_values(by='start_time_s', ascending=True)
+        # indices <- list of time stamps of amplitude peaks
+        time_series = temp_df['amplitude']
+        indices = find_peaks(time_series, plateau_size=1)[0]
```
Comment (tjlam) [inline audio-analysis/update-audio-peaks.py]: Does this find the global max or all local maxes?

File: audio-analysis/update-audio-peaks.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        for index, row in temp_df.iterrows():
+            if row['start_time_s'] not in indices:
+                temp_df.at[index, 'is_amplitude_peak'] = False
+            else:
+                temp_df.at[index, 'is_amplitude_peak'] = True
+        dfHelper.updateDBWithDataFromDF(temp_df, engine, column_name)
+    print('Done!')
```
Comment (tjlam) [inline audio-analysis/update-audio-peaks.py]: new line

**Review by tjlam — DISMISSED** [review]
Just some questions but it LGTM. Would be good addition to have in our model for sure. Would be interesting to query for the max amps and do a quick spot check. See how well it works

File: audio-analysis/update-audio-peaks.py (hunk 1 above)
Comment (mattlianje) [reply audio-analysis/update-audio-peaks.py:38]: just a little loading bar lib)

File: audio-analysis/update-audio-peaks.py (hunk 2 above)
Comment (mattlianje) [reply audio-analysis/update-audio-peaks.py]: local maxes)

File: audio-analysis/update-audio-peaks.py (hunk 3 above)
Comment (mattlianje) [reply audio-analysis/update-audio-peaks.py]: yessir)

**Comment by mattlianje** [comment]
> Just some questions but it LGTM. Would be good addition to have in our model for sure. Would be interesting to query for the max amps and do a quick spot check. See how well it works

@tjlam Should have started with with this but yes - visual inspections says it looks aight
image

**Review by tjlam — APPROVED** [review]
This is sick, Nice work :)))

**Pull request merged**